source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -165,7 +165,8 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
   switch (response->status) {
   case CheckStatus::OK: {
     if (config_->clearRouteCache() &&
-        (!response->headers_to_set.empty() || !response->headers_to_append.empty())) {
+        (!response->headers_to_set.empty() || !response->headers_to_append.empty() ||
+         !response->headers_to_remove.empty())) {
```

Comment:
Sorry why we need this? Could you give some inline comments here?

Reply:
Yeah great point. I'll add a comment here.